### PR TITLE
#1031: Bloom Filter - Migrated `bf.add`, `bf.reserve`,  `bf.exists` & `bf.info` to store_eval

### PR DESCRIPTION
**Pull request by apoorvyadav1111**
Fixes: #1031 #1110

For migration commands, please consider this a template on how to tackle the issue. I have divided the tasks into smaller goals and made an individual commit.
| Condition                                      | Return Value                                      |
|------------------------------------------------|---------------------------------------------------|
| Migrating eval to store eval | [ebba934](https://github.com/DiceDB/dice/pull/1117/commits/ebba9347ceac373e28ef8efea04a35b7c35bc99c)
|Modify unit test with new response format | [d4ae3eb](https://github.com/DiceDB/dice/pull/1117/commits/d4ae3eb4ab8859c6686bfef5bc7530e1ec80d507) |
| Migrate integration tests from async to resp| [81ff86e](https://github.com/DiceDB/dice/pull/1117/commits/81ff86e655583733065b6b198d4f732601a37014)
| Add/update integration in websocket| [c1764b4](https://github.com/DiceDB/dice/pull/1117/commits/c1764b42b67784da3d4dda444b8ce7f916e3233d)
| Add/update integration in HTTP | [42fd50c](https://github.com/DiceDB/dice/pull/1117/commits/42fd50c7fc594ea5bcb12130271f5b1b70e840e5)
| Ensure all integration test pass | [42fd50c](https://github.com/DiceDB/dice/pull/1117/commits/42fd50c7fc594ea5bcb12130271f5b1b70e840e5) |
| Update/Add docs for the commands | [afb541c](https://github.com/DiceDB/dice/pull/1117/commits/afb541c9ef84b70c23eef321db7efb862e2bd4dd) |

Checklist

- [x]  Migrated the evalXXX function with the latest definition
- [x]  Update or add unit tests for the new implementation.
- [x]  All unit tests pass successfully.
- [x]  Ensure all integration tests pass successfully.
- [x] Move Integration tests for the respective commands under the RESP integration tests directory from the Async directory
- [ ] Please validate that the documentation for the respective commands is up to date. If not then consider adding them.

### CODE REVIEW

**Comment by AshwinKul28** [comment]
Thanks for this PR @apoorvyadav1111 . Fantastic effort. Everything looks good and we can now reuse evalError and evalResponse functions in all store evals. We may not ask others to do the changes again if other PRs are already in the last stage, we will do it by ourselves once all commands are migrated. 

Also, I see your commit links are broken in the PR description for now, once it's added to master I believe it should work. 

Thanks again. I just have one comment, post resolution of that feel free to merge.

**Review by AshwinKul28 — APPROVED** [review]
LGTM

File: integration_tests/commands/async/object_test.go [hunk 1]
```diff
@@ -63,7 +64,7 @@ func TestObjectCommand(t *testing.T) {
 		},
 		{
 			name:       "Object Encoding check for bf",
-			commands:   []string{"BFADD bloomkey value1", "BFADD bloomkey value2", "OBJECT ENCODING bloomkey"},
```
[the remark this answers is not in the document]
Comment (apoorvyadav1111) [reply integration_tests/commands/async/object_test.go:66]: I have kept it here so that it can be migrated when Object commands are migrated.

**Comment by apoorvyadav1111** [comment]
Hi @AshwinKul28 , thanks for the review. It must be due to force-push, for the safer side, I will update the commits once the branch gets merged and will not be modified.

File: internal/eval/eval.go [hunk 2]
```diff
@@ -54,6 +54,22 @@ type EvalResponse struct {
 	Error  error       // Error holds any error that occurred during the operation. If no error, it will be nil.
 }
 
+//go:inline
+func makeEvalResult(result interface{}) *EvalResponse {
```
Comment (AshwinKul28) [inline internal/eval/eval.go:58]: Can we have a comment about this function?

File: internal/eval/eval.go [hunk 3]
```diff
@@ -54,6 +54,22 @@ type EvalResponse struct {
...
+	return &EvalResponse{
+		Result: result,
+		Error:  nil,
+	}
+}
+
+//go:inline
+func makeEvalError(err error) *EvalResponse {
```
Comment (AshwinKul28) [inline internal/eval/eval.go:66]: Can we have a comment about this function?

**Review by AshwinKul28 — CHANGES_REQUESTED** [review]

**Comment by apoorvyadav1111** [comment]
Hi @AshwinKul28 , Apologies I merged it before looking at the new commits, I can definitely raise a new PR right now and add those comments. I have updated the commit hashes for now.

File: internal/eval/eval.go (hunk 2 above)
Comment (apoorvyadav1111) [reply internal/eval/eval.go:58]: Hi @AshwinKul28 , Please check the #1186. I merged the PR eagerly I suppose. Apologies. We can use that PR for more updates If left in this PR.

File: internal/eval/eval.go (hunk 3 above)
Comment (apoorvyadav1111) [reply internal/eval/eval.go:66]: Hi @AshwinKul28 , Please check the #1186. I merged the PR eagerly I suppose. Apologies. We can use that PR for more updates If left in this PR.

**Pull request merged**